cookbooks/aws-parallelcluster-environment/recipes/config/mount_cx_fs.rb

```diff
@@ -10,15 +10,34 @@
 # or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
 # OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
 # limitations under the License.
+return if on_docker?
+efs_shared_dir_array = node['cluster']['efs_shared_dirs'].split(',')
+efs_fs_id_array = node['cluster']['efs_fs_ids'].split(',')
+efs_encryption_in_transit_array = node['cluster']['efs_encryption_in_transits'].split(',')
+efs_iam_authorization_array = node['cluster']['efs_iam_authorizations'].split(',')
 
-# Mount EFS directory with efs resource
+cx_shared_dir_array = []
+cx_efs_fs_id_array = []
+cx_efs_encryption_array = []
+cx_efs_iam_array = []
+
+# Identify the customer use filesystems and store their data in arrays for the EFS resource
+efs_shared_dir_array.each_with_index do |dir, index|
+  next if node['cluster']['internal_shared_dirs'].include?(dir)
```

Comment:
Correct me if I'm wrong: `efs_shared_dir_array` contains the customer shared storage already (ie: the storage defined in SharedStorage section in config file). So this filter here is to prevent the customer to mount a storage over an internal storage. Right? If this is the goal, note that this filter should be applied at the CLI level, if it's not already there.

Reply:
In the CLI, I am adding the internal storage using the add_shared_storage mechanism and filtering the data here

Reply:
This filter is just to distinguish which fses need to be mounted when, not dealing with overlap (which I agree needs to happen in the cli